### ISSUE TITLE
[FR] layout for float client

### ISSUE THREAD

**Issue by zdmgmail** [awesomeWM/awesome#3440]
Hope each tag can set layout for nofloat client, and another layout for float client (may be layout for float client  need restrict some move and resize) 

so will get those benifit:

- give small gap for nofloat client, big gap for float client will more beautiful
- use mod+j, mod+k for focus nofloat client, use alt+j, alt+k for float client will more convient
- they don't block each other. So new workflow can lesser change tag,  give opacity for float client,  then use a shortcut show and hide all float client

I know placement can work for some case, but not good as layout

**Comment by Elv13** [comment]
Well, if I understand what you ask, I don't think it needs any new APIs or changes to AwesomeWM. You can already get the floating clients and move them using code. That's exactly what the layouts do. So you can just implement your keybindings and use `:geometry()` to move the client the way you like them.

Tiled clients are not "special" in AwesomeWM. They are just normal clients with `floating = flase; maximized = false; minimized = false`. Then the layout code take those clients and move them around. They are still floating and "normal". It's just that when you drag them, the layout is recomputed each time. That give the illusion that they are bound to a grid, but they are not.

**Comment by zdmgmail** [comment]
@Elv13  can do , but change each layout seems difficult ...  and keypoint is float client should auto set geometry by layout , not about keybinding

**Comment by Elv13** [comment]
I may not understand exactly what you ask, but I don't think you need to change every layout to achieve "things". Floating clients are not tied to the tiled layout. You can implement your own geometry logic on top of it. That's what `awful.placement` does. I think what you want to a little more far reaching than `awful.placement`, but the mechanic is the same.

**Comment by actionless** [comment]
yup, i think you need to add a `rule` for those clients you want to affect, in which you need to add a custom `placement` function, or try combining already existing ones to reach the desired effect:
ex. https://github.com/actionless/awesome_config/blob/253c3ab4f21b7683f9ad00db33e05ffc2c24c62c/config/rules.lua#L45-L72

regarding universal hotkeys - you just need to add `if` in the keybindings:
ex. https://github.com/actionless/awesome_config/blob/253c3ab4f21b7683f9ad00db33e05ffc2c24c62c/config/keys.lua#L906-L923 ,
https://github.com/actionless/awesome_config/blob/253c3ab4f21b7683f9ad00db33e05ffc2c24c62c/config/keys.lua#L771-L797

i think in my local config i have all/most of the things resolved which you mention in the original question
